lib/galaxy/webapps/galaxy/services/datasets.py

```diff
@@ -367,6 +375,50 @@ def show_inheritance_chain(
 
         return DatasetInheritanceChain(__root__=result)
 
+    def get_drs_object(
+        self,
+        trans: ProvidesHistoryContext,
+        object_id: EncodedDatabaseIdField,
+        request_url: URL,
+        hda_ldda: DatasetSourceType = DatasetSourceType.hda,
+    ) -> DrsObject:
+        decoded_dataset_id = self.decode_id(object_id)
+        dataset_instance = self.dataset_manager_by_type[hda_ldda].get_accessible(decoded_dataset_id, trans.user)
+        if not trans.app.security_agent.dataset_is_public(dataset_instance.dataset):
+            # Only public datasets may be access as DRS datasets currently
+            raise galaxy_exceptions.ObjectNotFound("Cannot find a public dataset with specified object ID.")
+
+        # TODO: issue warning if not being served on HTTPS @ 443 - required by the spec.
+        self_uri = f"drs://drs.{request_url.components.netloc}/{object_id}"
+        checksums: List[Checksum] = []
+        for dataset_hash in dataset_instance.dataset.hashes:
+            if dataset_hash.extra_files_path:
+                continue
+            type = dataset_hash.hash_function
+            checksum = dataset_hash.hash_value
+            checksums.append(Checksum(type=type, checksum=checksum))
+
+        if len(checksums) == 0:
+            # fire off a checksuming task and return a 202?
+            checksums.append(Checksum(type="md5", checksum="dummysum"))
```

Comment:
maybe raise an exception for now ?

Reply:
Rebased this on top of https://github.com/galaxyproject/galaxy/pull/14576 and it now fires off a Celery request to MD5 the dataset and returns a 202. I've updated the DRS client to handle the 202 responses.